pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go

```diff
@@ -98,3 +120,77 @@ func TestGetBranches(t *testing.T) {
 		})
 	}
 }
+
+/*
+ * Testing for getMergeRequests
+ */
+func TestGetMergeRequests(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		client          graphql.Client
+		expectedErr     error
+		expectedNodeLen int
+	}{
+		{
+			desc:            "empty mergeRequestData",
+			client:          &mockClient{},
+			expectedErr:     nil,
+			expectedNodeLen: 0,
+		},
+		{
+			desc:        "produce error in client",
+			client:      &mockClient{err: true, errString: "An error has occurred"},
+			expectedErr: errors.New("An error has occurred"),
+		},
+		{
+			desc: "valid mergeRequestData",
+			client: &mockClient{
+				maxPages: 1,
+				MergeRequests: getMergeRequestsProjectMergeRequestsMergeRequestConnection{
+					PageInfo: getMergeRequestsProjectMergeRequestsMergeRequestConnectionPageInfo{
+						HasNextPage: true,
+					},
+					Nodes: []MergeRequestNode{
+						{
+							SourceBranch: "main",
+						},
+						{
+							SourceBranch: "dev",
+						},
+						{
+							SourceBranch: "feature",
+						},
+					},
+				},
+			},
+			expectedErr:     nil,
+			expectedNodeLen: 3,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			factory := Factory{}
+			defaultConfig := factory.CreateDefaultConfig()
+			settings := receivertest.NewNopCreateSettings()
+			gls := newGitLabScraper(context.Background(), settings, defaultConfig.(*Config))
+			const state MergeRequestState = "merged"
+
+			var wg sync.WaitGroup
+			ch := make(chan []MergeRequestNode, 1)
+
+			wg.Add(1)
+			gls.getMergeRequests(context.Background(), tc.client, "projectPath", state, ch, &wg)
+			wg.Wait()
+			close(ch)
+
+			if tc.expectedErr != nil {
+				assert.Equal(t, len(ch), 1)
+			} else {
+				assert.Equal(t, len(ch), 1)
+				for data := range ch {
+					assert.Equal(t, len(data), tc.expectedNodeLen)
```

Comment:
This should also be renamed to total mr count (idk how to suggest a code change)

Reply:
Changed to expectedMergeRequestLength

I think this is what you mean.